projects/addon-editor/components/edit-link/edit-link.component.ts

```diff
@@ -101,33 +131,56 @@ export class TuiEditLinkComponent {
         this.url = ``;
     }
 
-    private makeUrl(): string {
-        const selection = this.documentRef.getSelection();
+    private makeDefaultPrefix(): TuiLinkPrefix {
+        const a = this.getAnchorElement();
 
-        return selection ? this.getHref(selection) : ``;
+        if (a) {
+            return (!a.getAttribute(`href`) && a.getAttribute(`id`)) ||
+                a.getAttribute(`href`)?.startsWith(`#`)
+                ? HASH_PREFIX
+                : HTTP_PREFIX;
+        }
+
+        return HTTP_PREFIX;
     }
 
-    private getHref({focusNode}: Selection): string {
-        if (!focusNode?.parentElement) {
+    private detectAnchorMode(): boolean {
+        return this.getAnchorElement()?.href ? false : this.href.startsWith(HASH_PREFIX);
```

Comment:
`return !this.getAnchorElement()?.href && this.href.startsWith(HASH_PREFIX);`

Reply:
fixed